controllers/metal3datatemplate_controller.go

```diff
@@ -54,57 +69,65 @@ func (r *Metal3DataTemplateReconciler) Reconcile(req ctrl.Request) (_ ctrl.Resul
 	metadataLog := r.Log.WithName(dataTemplateControllerName).WithValues("metal3-datatemplate", req.NamespacedName)
 
 	// Fetch the Metal3DataTemplate instance.
-	capm3Metadata := &capm3.Metal3DataTemplate{}
+	capm3DataTemplate := &capm3.Metal3DataTemplate{}
 
-	if err := r.Client.Get(ctx, req.NamespacedName, capm3Metadata); err != nil {
+	if err := r.Client.Get(ctx, req.NamespacedName, capm3DataTemplate); err != nil {
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
 		return ctrl.Result{}, err
 	}
-	helper, err := patch.NewHelper(capm3Metadata, r.Client)
+	helper, err := patch.NewHelper(capm3DataTemplate, r.Client)
 	if err != nil {
 		return ctrl.Result{}, errors.Wrap(err, "failed to init patch helper")
 	}
 	// Always patch capm3Machine exiting this function so we can persist any Metal3Machine changes.
 	defer func() {
-		err := helper.Patch(ctx, capm3Metadata)
+		err := helper.Patch(ctx, capm3DataTemplate)
 		if err != nil {
-			metadataLog.Info("failed to Patch capm3Metadata")
+			metadataLog.Info("failed to Patch capm3DataTemplate")
 		}
 	}()
 
-	// Fetch the Cluster. Ignore an error if the deletion timestamp is set
-	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, capm3Metadata.ObjectMeta)
-	if capm3Metadata.ObjectMeta.DeletionTimestamp.IsZero() {
-		if err != nil {
-			metadataLog.Info("Metal3DataTemplate is missing cluster label or cluster does not exist, Requeuing")
-			return ctrl.Result{}, nil
-		}
-		if cluster == nil {
-			metadataLog.Info(fmt.Sprintf("This metadata is not yet associated with a cluster using the label %s: <name of cluster>", capi.ClusterLabelName))
-			return ctrl.Result{}, nil
-		}
+	cluster := &capi.Cluster{}
+	key := client.ObjectKey{
+		Name:      capm3DataTemplate.Spec.ClusterName,
+		Namespace: capm3DataTemplate.Namespace,
 	}
 
-	if cluster != nil {
-		metadataLog = metadataLog.WithValues("cluster", cluster.Name)
+	if capm3DataTemplate.ObjectMeta.Labels == nil {
+		capm3DataTemplate.ObjectMeta.Labels = make(map[string]string)
+	}
+	capm3DataTemplate.ObjectMeta.Labels[capi.ClusterLabelName] = capm3DataTemplate.Spec.ClusterName
+	capm3DataTemplate.ObjectMeta.Labels[capi.ProviderLabelName] = "infrastructure-metal3"
```

Comment:
Is there any tests which is checking whether the label's are set correctly?

Reply:
The labels are actually not needed, so I will remove them